xbmc/platform/linux/input/LIRC.h

```diff
@@ -24,28 +24,33 @@
 #include <atomic>
 
 #include "system.h"
+#include "input/remote/IRemoteControl.h"
 #include "threads/Thread.h"
 #include "threads/Event.h"
 
-class CRemoteControl :  CThread
+class CRemoteControl : public IRemoteControl, CThread
 {
 public:
   CRemoteControl();
   ~CRemoteControl() override;
-  void Initialize();
-  void Disconnect();
-  void Reset();
-  void Update();
-  unsigned short GetButton();
+  void Initialize() override;
+  void Disconnect() override;
+  void Reset() override;
+  void Update() override;
+  unsigned short GetButton() const override;
   /*! \brief retrieve the time in milliseconds that the button has been held
    \return time in milliseconds the button has been down
    */
-  unsigned int GetHoldTime() const;
-  void SetDeviceName(const std::string& value);
-  void SetEnabled(bool value);
-  bool IsInUse() const { return m_used; }
-  bool IsInitialized() const { return m_bInitialized; }
-  void AddSendCommand(const std::string& command);
+  unsigned int GetHoldTime() const override;
+  void SetDeviceName(const std::string& value) override;
+  void SetEnabled(bool value) override;
+  bool IsInUse() const override { return m_used; }
+  bool IsInitialized() const override { return m_bInitialized; }
+  void AddSendCommand(const std::string& command) override;
+  std::string GetMapFile() override { return "Lircmap.xml"; }
```

Comment:
Can you move this definition to the .cpp file and make `"Lircmap.xml"` a #define?